scripts/assign_workflow_preseqlab.py

```diff
@@ -1,28 +1,79 @@
 #!/usr/bin/env python
+import platform
+
+import time
+from pyclarity_lims.entities import Step
+from requests import HTTPError
+
 from EPPs.common import StepEPP, get_workflow_stage, find_newest_artifact_originating_from
 
 
 class AssignWorkflowPreSeqLab(StepEPP):
+
+    @staticmethod
+    def _finish_step(step, try_count=1):
+        """
+        This function will try to advance a step three time waiting for a ongoing program to finish.
+        It waits for 5 seconds in between each try
+        """
+        try:
+            step.get(force=True)
+            step.advance()
+        except HTTPError as e:
+            if try_count < 3 and str(e) == '400: Cannot advance a step that has an external program queued, ' \
```

Comment:
I think this will only try twice. Might be good to add a test for this as well

Reply:
No it will try  three time as it is.
I've added test now